### PR TITLE
shell.nix: Add more runtime dependencies, fix opengl dependencies and wrapper

### PR DESCRIPTION
**Pull request by timor**
Required for vocabs:
- ogg
- ogg.vorbis
- compression.zlib.ffi

Theoretically, we would need to include every library that has been declared with `LIBRARY:` to make sure all vocabs with ffi work out-of-the box.  This may result in an unacceptable nix closure size though.
Possibly related: #641 

cc @spacefrogg

### CODE REVIEW

**Comment by timor** [comment]
@spacefrogg
I basically messed up every line of code twice on this PR.  Maybe if you have some time you could take a look at this.

**Comment by spacefrogg** [comment]
As I see it right now, this asks for an interface.
Just like for python or other languages, you should define a set of "packages" that the user can activate in some kine of `factorWithPackages`. Right now, they would not actually draw in a factor vocab but set up the necessary dependencies such that the resulting factor can work with those vocabs. This solves 2 1/2 problems:
- It gives the user a list of viable options and resolves her dependency issues by creating the relation between the factor vocab she wants to use and the necessary setup that has to be performed.
- It prepares factor to actually integrate off-source vocabs.

**Comment by spacefrogg** [comment]
It also shifts the responsibility for the closure size to the user, where it belongs.

**Comment by timor** [comment]
> As I see it right now, this asks for an interface.
> Just like for python or other languages, you should define a set of "packages" that the user can activate in some kind of `factorWithPackages`.
That we already have in the not-yet-upstreamed new factor derivation. The build helper code there still has to be changed to use this repo's shell.nix.

The other question is whether the changes here look correct in fixing the opengl package dependency issues and the changes to the `wrapFactor` helper...

**Comment by spacefrogg** [comment]
It works.

**Comment by timor** [comment]
I used this for a while now.  There is still an impurity problem, which happens when Factor is started from an Emacs that has been built against a different version of glib/gtk.

Nevertheless, it would be nice if this could get merged to fix compatibility with recent NixOS versions.

**Comment by spacefrogg** [comment]
LGTM.

File: shell.nix [hunk 1]
```diff
@@ -31,6 +35,11 @@ in
     [ -n "$1" ] || { printf "Usage: wrapFactor <factor-root>" ; return; }
     local root="$(realpath $1)"
     local binary="''${root}/factor"
+    local wrapped="''${root}/.factor-wrapped"
+    # Remove the wrapped binary if a new VM has been compiled
+    ${lib.getBin file}/bin/file $binary |grep ELF >/dev/null && rm -f "$wrapped"
```
Comment (mrjbq7) [inline shell.nix:40]: does ``$binary`` need to be in quotes in case ``root`` has spaces in it?

Comment (timor) [reply shell.nix:40]: I think you are correct.

**Comment by mrjbq7** [comment]
Merged and fixed in 1928e6094ea644b24e511e5c21036729b017f98d.

**Comment by timor** [comment]
Thanks!

**Pull request merged**